src/main/java/HTTP/HTTPToken.java

```diff
@@ -0,0 +1,45 @@
+package HTTP;
+import lombok.SneakyThrows;
+import org.apache.http.impl.client.CloseableHttpClient;
+
+import static HTTP.HTTPVariables.URL_FOR_READ_TOKEN;
+import static HTTP.HTTPVariables.URL_FOR_WRITE_TOKEN;
+
+public class HTTPToken {
+
+    private static String writeToken;
+    private static String readToken;
+
+    HTTPBasicAuthClient httpBasicAuthClient = new HTTPBasicAuthClient();
+
+    @SneakyThrows
+    private String getOauthToken(CloseableHttpClient httpClient, String URL) {
+        HTTPResponseDTO httpResponseDTO = httpBasicAuthClient.callPostReturnHTTPResponseData(httpClient, URL);
+        return httpResponseDTO.getBody().getString("access_token");
+    }
+
+    @SneakyThrows
+    private HTTPToken() {
+        CloseableHttpClient httpClient = httpBasicAuthClient.getHttpClientWithBasicAuthCredentials();
```

Comment:
I also don't see the reason to create instance of CloseableHttpClient on this level as it can be done in HttpClient class

Reply:
I created an instance of `CloseableHttpClient ` because I needed to close the client, at the 26 row, right after he did useful stuff.

Let's discuss it at a meeting.

Reply:
I moved this instance into the `ClientBasicAuth.getOauthToken` method.